swap/src/bitcoin/wallet.rs

```diff
@@ -344,13 +345,20 @@ where
             fee_rate,
             sats_per_vbyte
         );
+
         if sats_per_vbyte < min_relay_fee.as_sat() {
             tracing::warn!(
                 "Estimated fee of {} is smaller than the min relay fee, defaulting to min relay fee {}",
                 sats_per_vbyte,
                 min_relay_fee.as_sat()
             );
             Ok(min_relay_fee)
+        } else if sats_per_vbyte > MAX_TX_FEE {
+            tracing::warn!(
+                "Hard bound of max transaction fees reached. Falling back to: {} sats",
+                MAX_TX_FEE
+            );
+            Ok(bitcoin::Amount::from_sat(MAX_TX_FEE))
```

Comment:
Note: upper bound could also be defined as a percentage of the amount being swapped.